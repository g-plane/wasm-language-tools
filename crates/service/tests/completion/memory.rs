use super::*;
use insta::assert_json_snapshot;
use lsp_types::{Position, Uri};
use wat_service::LanguageService;

#[test]
fn export_desc_memory() {
    let uri = "untitled:test".parse::<Uri>().unwrap();
    let source = "
(module
    (export \"\" (memory ))
    (memory $memory (data))
)
";
    let mut service = LanguageService::default();
    service.commit_file(uri.clone(), source.into());
    let response = service.completion(create_params(uri, Position::new(2, 23)));
    assert_json_snapshot!(response);
}

#[test]
fn export_desc_memory_following_dollar() {
    let uri = "untitled:test".parse::<Uri>().unwrap();
    let source = "
(module
    (export \"\" (memory $))
    (memory $memory (data))
)
";
    let mut service = LanguageService::default();
    service.commit_file(uri.clone(), source.into());
    let response = service.completion(create_params(uri, Position::new(2, 24)));
    assert_json_snapshot!(response);
}

#[test]
fn export_desc_memory_incomplete() {
    let uri = "untitled:test".parse::<Uri>().unwrap();
    let source = "
(module
    (export \"\" (memory $m))
    (memory $memory (data))
)
";
    let mut service = LanguageService::default();
    service.commit_file(uri.clone(), source.into());
    let response = service.completion(create_params(uri, Position::new(2, 25)));
    assert_json_snapshot!(response);
}
