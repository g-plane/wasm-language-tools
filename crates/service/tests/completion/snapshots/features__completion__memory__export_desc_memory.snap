---
source: crates/service/tests/completion/memory.rs
expression: response
---
[
  {
    "label": "$memory",
    "kind": 6
  }
]
